File: src/foam/core/ContextAwareSupport.java
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

package foam.core;

public class ContextAwareSupport
  implements ContextAware
{
  protected X x_;

  public X    getX() { return x_; }
  public void setX(X x) { x_ = x; }
}

File: src/foam/core/ContextAwareSupport.js
/**
 * @license
 * Copyright 2017 The FOAM Authors. All Rights Reserved.
 * http://www.apache.org/licenses/LICENSE-2.0
 */

foam.CLASS({
  package: 'foam.core',
  name: 'ContextAwareSupport',

  implements: [
    'foam.core.ContextAware'
  ],

  properties: [
    {
      class: 'Object',
      name: 'x',
      javaType: 'foam.core.X',
      javaFactory: `
        return EmptyX.instance();
      `
    }
  ]
})
